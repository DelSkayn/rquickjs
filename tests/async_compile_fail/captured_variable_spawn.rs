use rquickjs::{async_with, prelude::*, AsyncContext, AsyncRuntime};

pub async fn test() {
    let rt = AsyncRuntime::new().unwrap();
    let ctx = AsyncContext::full(&rt).await.unwrap();

    let mut var = 1u32;
    let var_ref = &mut var;
    async_with!(ctx => |ctx|{
        let func = Func::from(MutFn::from(move ||{
            *var_ref += 1;
        }));
        ctx.globals().set("t",func).unwrap();
    })
    .await
}

fn main() {}
