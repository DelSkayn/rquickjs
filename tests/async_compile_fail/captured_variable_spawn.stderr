error[E0597]: `var` does not live long enough
  --> tests/async_compile_fail/captured_variable_spawn.rs:8:19
   |
7  |     let mut var = 1u32;
   |         ------- binding `var` declared here
8  |     let var_ref = &mut var;
   |                   ^^^^^^^^ borrowed value does not live long enough
...
13 |         ctx.globals().set("t",func).unwrap();
   |         --------------------------- argument requires that `var` is borrowed for `'static`
...
16 | }
   | - `var` dropped here while still borrowed
